src/test/java/org/apache/maven/plugins/resources/stub/MavenProjectBuildStub.java

```diff
@@ -291,24 +291,18 @@ private void createFiles( String parent, String testparent )
         }
     }
 
-    private void populateFile( File file )
+    private void populateFile( File file ) throws IOException
     {
-        FileOutputStream outputStream;
         String data = dataMap.get( file.getName() );
 
         if ( ( data != null ) && file.exists() )
         {
-            try
+            try ( FileOutputStream outputStream = new FileOutputStream( file ) )
             {
-                outputStream = new FileOutputStream( file );
                 outputStream.write( data.getBytes() );
                 outputStream.flush();
                 outputStream.close();
```

Comment:
The try-with-resources should handle the `close()` - this line is now unnecessary.

Reply:
yup no need of flush neither